sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -136,11 +145,45 @@ void scheduleMessage() {
             .verifyComplete();
     }
 
+    /**
+     * Verifies that we can schedule and receive multiple messages.
+     */
+    @Test
+    void testSendMultipleSceduledMessageAndReceive() {
+        // Arrange
+        final String messageId1 = UUID.randomUUID().toString();
+        final String messageId2 = UUID.randomUUID().toString();
+        String contents = "Some-contents";
+        final ServiceBusMessage message1 = TestUtils.getServiceBusMessage(contents, messageId1, 0);
+        final ServiceBusMessage message2 = TestUtils.getServiceBusMessage(contents, messageId2, 0);
+        final Instant scheduledEnqueueTime = Instant.now().plusSeconds(1);
+
+        sender.scheduleMessage(message1, scheduledEnqueueTime)
+            .block();
+        sender.scheduleMessage(message2, scheduledEnqueueTime)
+            .block(Duration.ofSeconds(4));
+
+        // Assert & Act
+        String finalContents = contents;
+        StepVerifier.create(receiveDeleteModeReceiver.receive().take(2))
+            .assertNext(receivedMessage -> {
+                Assertions.assertArrayEquals(finalContents.getBytes(), receivedMessage.getBody());
+                Assertions.assertTrue(receivedMessage.getProperties().containsKey(MESSAGE_TRACKING_ID));
+                Assertions.assertEquals(messageId1, receivedMessage.getProperties().get(MESSAGE_TRACKING_ID));
```

Comment:
I wouldn't do this equality check. If any prior tests fail, those old messages will be in the queue, so it'll be received first. And, this test won't delete the matching messageId and fail. That's what some of our live tests are encountering.

Reply:
Same with other instances of this assertion.